returnn/compile.py

```diff
@@ -105,3 +107,64 @@ def hash(cls, kwargs):
         c = copy.copy(kwargs)
         del c["verbosity"]
         return super().hash(c)
+
+
+class CompileNativeOpJob(Job):
+    """
+    Compile a RETURNN native op into a shared object file.
+    """
+
+    def __init__(
+        self,
+        native_op,
+        returnn_python_exe=None,
+        returnn_root=None,
+    ):
+        """
+        :param str native_op: Name of the native op to compile (e.g. NativeLstm2)
+        :param Path|str returnn_python_exe: file path to the executable for running returnn (python binary or .sh)
+        :param Path|str returnn_root: file path to the RETURNN repository root folder
+        """
+        self.native_op = native_op
+        self.returnn_python_exe = (
+            returnn_python_exe
+            if returnn_python_exe is not None
+            else gs.RETURNN_PYTHON_EXE
+        )
+        self.returnn_root = (
+            returnn_root if returnn_root is not None else gs.RETURNN_ROOT
+        )
+
+        self.out_op = self.output_path("op.so")
```

Comment:
I think here it might be better if the file is actually named like the op, so using `native_op` instead of `"op.so"`